scapy/tools/automotive/isotpscanner.py

```diff
@@ -0,0 +1,157 @@
+        description="Scan for open ISOTP-Sockets.",
+        formatter_class=RawTextHelpFormatter,
+        prog="ISOTPScanner",
+        usage="\tISOTPScanner.py interface startID endID [-flags] \n"
+              "\tpython -m scapy.tools.automotive.isotpscanner interface "
+              "startID endID [-flags]",
+        epilog="Example of use:\n\n"
+               "Python2 or Windows:\n"
+               "python2 -m scapy.tools.automotive.isotpscanner "
+               "\"can.interface.Bus(bustype='pcan', channel='PCAN_USBBUS1', "
+               "bitrate=250000)\" 0 100 \n"
+               "python2 -m scapy.tools.automotive.isotpscanner "
+               "\"can.interface.Bus(bustype='vector', channel=0, "
+               "bitrate=250000)\" 0 100 \n"
+               "python2 -m scapy.tools.automotive.isotpscanner "
+               "\"can.interface.Bus(bustype='socketcan', channel='can0', "
+               "bitrate=250000)\" 0 100 \n\n"
+               "Python3 on Linux:\n"
+               "python3 -m scapy.tools.automotive.isotpscanner can0 0 100")
+    parser.add_argument("interface", type=str,
+                        help="CAN interface for the scan.\n"
+                             "Depends on used interpreter and system,\n"
+                             "see examples below. Any python-can interface can"
+                             " be provided. Please see: "
+                             "https://python-can.readthedocs.io for further "
+                             "interface examples.")
+    parser.add_argument("startID", type=lambda x: int(x, 16),
+                        help="Start scan at this identifier (hex)"),
+    parser.add_argument("endID", type=lambda x: int(x, 16),
+                        help="End scan at this identifier (hex)")
+    parser.add_argument("-n", "--noise_listen_time", type=int, default=2,
+                        help="Set seconds listening for noise before scan.")
+    parser.add_argument("-t", "--sniff_time", type=int, default=100,
+                        help="Set the duration, a sniff is waiting for a "
+                             "flow-control response. In milliseconds.")
+    parser.add_argument("-e", "--extended", action="store_true",
+                        help="Include ISOTP extended addressing to scan.")
+    parser.add_argument("-eo", "--extended_only", action="store_true",
+                        help="Scan only with ISOTP extended addressing.")
+    parser.add_argument("-p", "--piso", action="store_true",
+                        help="Print 'Copy&Paste'-ready ISOTPSockets.")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Display information during scan.")
+
+    args = parser.parse_args()
+
+    scan_interface = args.interface
+    interface_string = ""
+    if "can.interface.Bus" in scan_interface:
+        if PYTHON_CAN:
+            import can  # noqa: 401
+            try:
+                scan_interface = eval(scan_interface)
```

Comment:
This looks way to dangerous and I don't see why it would be required :/

Reply:
I agree. It is very likely exploitable. 
I don't know another way to do the following from a command line.
On windows or python2 I have to use python-can to access a can-interface. 
Here is the usage example:
```
Python2 or Windows:
python2 -m scapy.tools.automotive.isotpscanner "can.interface.Bus(bustype='pcan', channel='PCAN_USBBUS1', bitrate=250000)" 0 100 
python2 -m scapy.tools.automotive.isotpscanner "can.interface.Bus(bustype='vector', channel=0, bitrate=250000)" 0 100 
python2 -m scapy.tools.automotive.isotpscanner "can.interface.Bus(bustype='socketcan', channel='can0', bitrate=250000)" 0 100 

Python3 on Linux:
python3 -m scapy.tools.automotive.isotpscanner can0 0 100
```
python-can supports a huge variety of interfaces:
https://python-can.readthedocs.io/en/master/interfaces.html

I would like to have an easy way to specify any of this interfaces for a scan. 

I need to create a python-can `Bus` object in order to use it with a `CANSocket`.
Do you know a safer way to do this?

Reply:
removed